src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2726,14 +1796,9 @@ bool processOnChangeConditionForSubscription
   Restriction                   res;
   ContextElementResponseVector  rawCerV;
 
-  /* Include scopes for entitiesQuery() if q filter is not empty */
```

Comment:
The right way of doing this is using "Scope inyection", i.e. recover the deleted code, as you have done in other placess. You would need to do the parse() step in order to "prepare" it for entitiesQuery().

In addition, that would allow to avoid stringFilterP and to go back to the old call to entitiesQuery()... That `NULL, NULL, NULL, "", false, false, "v1") is very hard to follow:)